dialects/hive/src/Database/Sql/Hive/Parser.hs

```diff
@@ -463,17 +463,21 @@ alterPartitionP = do
             ]
 
 
-setP :: Parser Range
+setP :: Parser (SetProperty Range)
 setP = do
     s <- Tok.setP
-    e <- option s $ choice
-        [ Tok.propertyNameP >> Tok.equalP >> last <$> propertyValueP
-        , Tok.symbolP "-" >> Tok.keywordP "v"
-        ]
-    return $ s <> e
-  where
-    propertyValueP = P.many1 Tok.notSemicolonP
+    option (PrintProperties s "") $ choice $
+      [ do
+        _ <- Tok.minusP >> Tok.keywordP "v"
```

Comment:
why do you need the monadic style here? I think you can get away with applicative parsing only?
```
Tok.minusP >> Tok.keywordP "v" *> pure (PrintProperties s "-v")
```